Code_Exercises/ND_Range_Kernel/solution.cpp

```diff
@@ -29,22 +29,25 @@ TEST_CASE("range_kernel_with_item", "nd_range_kernel_solution") {
   try {
     auto gpuQueue = sycl::queue{sycl::gpu_selector_v};
 
-    auto bufA = sycl::buffer{a, sycl::range{dataSize}};
-    auto bufB = sycl::buffer{b, sycl::range{dataSize}};
-    auto bufR = sycl::buffer{r, sycl::range{dataSize}};
-
-    gpuQueue.submit([&](sycl::handler& cgh) {
-      sycl::accessor accA{bufA, cgh, sycl::read_only};
-      sycl::accessor accB{bufB, cgh, sycl::read_only};
-      sycl::accessor accR{bufR, cgh, sycl::write_only};
-
-      cgh.parallel_for<vector_add_1>(
-          sycl::range{dataSize}, [=](sycl::item<1> itm) {
-            auto globalId = itm.get_id();
-            accR[globalId] = accA[globalId] + accB[globalId];
-          });
-    });
+    auto ptrA = sycl::malloc_device<int>(dataSize, gpuQueue);
+    auto ptrB = sycl::malloc_device<int>(dataSize, gpuQueue);
+    auto ptrR = sycl::malloc_device<int>(dataSize, gpuQueue);
 
+    gpuQueue.memcpy(ptrA, a, sizeof(int) * dataSize).wait();
```

Comment:
I don't really like the use of `.wait()` on every single async call. Would it be overkill to use `sycl::event`s to handle dependencies? It depends on the order that you will be presenting these on the day. Will synchronisation with events have been covered at this point? If not, then maybe some calls to `gpuQueue.wait()` at various points (between memcpys and kernel execution. `parallel_for(...).wait()` is OK IMO).